presto-parser/src/main/java/com/facebook/presto/sql/tree/IntegerLiteral.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.sql.tree;
+
+import com.facebook.presto.sql.parser.ParsingException;
+
+import java.util.Objects;
+import java.util.Optional;
+
+import static java.util.Objects.requireNonNull;
+
+public class IntegerLiteral
```

Comment:
As we discussed a few weeks ago, we shouldn't introduce a new AST node. The current `LongLiteral` should serve for all integer types (maybe with a change of name). Values should take the narrowest type the fit in (i.e., 1 => SMALLINT, 100000 => INTEGER, 100000000000 => BIGINT).